scio-core/src/main/scala/com/spotify/scio/runners/dataflow/DataflowContext.scala

```diff
@@ -63,31 +62,17 @@ case object DataflowContext extends RunnerContext {
 
     val classPathJars = classLoader.asInstanceOf[URLClassLoader]
       .getURLs
-      .map(url => new File(url.toURI).getCanonicalPath)
+      .map(url => new File(url.toURI))
+      .filter(_.exists())
+      .groupBy(_.getName)
+      .mapValues(_.head)
+      .values
+      .map(_.getCanonicalPath)
       .filter(p => !p.startsWith(javaHome) && p != userDir)
-      .toList
-
-    // fetch jars from classpath jar's manifest Class-Path if present
-    val manifestJars =  classPathJars
```

Comment:
Hi @ravwojdyla! @nevillelyh told me that you might know the main reason for this.